src/js/hooks/productSupplier/form/useProductSupplierForm.js

```diff
@@ -118,18 +144,30 @@ const useProductSupplierForm = () => {
       zodResolver(validationSchema(values))(values, context, options),
   });
 
+  useCalculateEachPrice({ control, setValue });
+
+  /* immediately trigger validation on preferenceType fields
+  when any values on defaultPreferenceType have changed */
+  const defaultPreferenceType = useWatch({ control, name: 'defaultPreferenceType' });
+  useEffect(() => {
+    trigger('defaultPreferenceType.preferenceType');
+  }, [defaultPreferenceType]);
```

Comment:
this was the point, we want to trigger this validation when any of the values in the default preference type subsection changes.
I tested dit with console logs on this useEffect and everything seems to work as expected.

Reply:
I know, but I meant that even if none values change, and something else triggers the re-render, this `useEffect` would anyway be triggered, because a reference of the object would change.